src/main/java/com/hedera/hashgraph/sdk/consensus/ConsensusClient.java

```diff
@@ -0,0 +1,131 @@
+package com.hedera.hashgraph.sdk.consensus;
+
+import com.google.common.collect.Iterators;
+import com.hedera.hashgraph.sdk.TimestampHelper;
+import com.hedera.mirror.api.proto.ConsensusServiceGrpc;
+import com.hedera.mirror.api.proto.ConsensusTopicQuery;
+import com.hedera.mirror.api.proto.ConsensusTopicResponse;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import io.grpc.CallOptions;
+import io.grpc.ClientCall;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.ClientCalls;
+import io.grpc.stub.StreamObserver;
+
+public class ConsensusClient implements AutoCloseable {
+    private final ManagedChannel channel;
+
+    private final HashMap<TopicId, Subscription> subscriptions = new HashMap<>();
+
+    public ConsensusClient(String endpoint) {
+        channel = ManagedChannelBuilder.forTarget(endpoint)
+            .usePlaintext()
+            .build();
+    }
+
+    public void subscribe(TopicId topic, TopicListener listener) {
+        subscriptions.computeIfAbsent(topic, _topic -> new Subscription(startStreamingCall(topic)))
+            .listeners
+            .add(listener);
+    }
+
+    private ClientCall<ConsensusTopicQuery, ConsensusTopicResponse> startStreamingCall(TopicId topic) {
+        final ClientCall<ConsensusTopicQuery, ConsensusTopicResponse> call =
+            channel.newCall(ConsensusServiceGrpc.getSubscribeTopicMethod(), CallOptions.DEFAULT);
+
+        final ConsensusTopicQuery topicQuery = ConsensusTopicQuery.newBuilder()
+            .setTopicID(topic.toProto())
+            .build();
+
+        ClientCalls.asyncServerStreamingCall(call, topicQuery, new StreamObserver<ConsensusTopicResponse>() {
+            @Override
+            public void onNext(ConsensusTopicResponse value) {
+                subscriptions.get(topic)
+                    .listeners
+                    .forEach(listener -> listener.onMessage(new ConsensusMessage(topic, value)));
+            }
+
+            @Override
+            public void onError(Throwable t) {
```

Comment:
This is a terminating error and we should have some backoff retry logic.

Reply:
After discussion, we just want to trigger an error handler and not have retry for now.